In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt

# 우리 데이터 QC

In [2]:
read_count = pd.read_csv('read_count.csv', sep=',', engine='python', encoding='cp949')
TPM_count = pd.read_csv('TPM_count.csv', sep=',', engine='python', encoding='cp949')

In [3]:
read_count

Name    Description  FN42  FN45  FN48  FN50  FN51  FN52  \
0       ENSG00000227232.5         WASH7P    30    91     5    38    20    55   
1       ENSG00000241860.7  RP11-34P13.13    27    27    55    46    34    16   
2       ENSG00000279457.4         WASH9P    19    42    20    88   110    43   
3      ENSG00000228463.10     AP006222.1    56    90     0    44    60    17   
4      ENSG00000237094.12   RP4-669L17.4    63    44    51    29    62    49   
...                   ...            ...   ...   ...   ...   ...   ...   ...   
20866   ENSG00000212907.2        MT-ND4L   475   273   412   556   266   396   
20867   ENSG00000198886.2         MT-ND4  2420  1268  2198  1929  1418  1781   
20868   ENSG00000198786.2         MT-ND5  1326  1232  1994  2134  1356  1452   
20869   ENSG00000198695.2         MT-ND6   156    68   108   119    69   148   
20870   ENSG00000198727.2         MT-CYB  4050  2420  5460  5708  2706  3933   

       FN54  FN56  ...  CFN03  CFN04  CFN05  CFN09  CFN11  RCFN01  RCFN06  \
0        37    22  ...     28     54     86      0      0       0       0   
1        44    62  ...      5     40      2      0     58     156       0   
2        16    14  ...    110     42     54    132     59     223       0   
3        70    58  ...      0     18     24      0     16       0       0   
4        47    75  ...     37     22      1     43     62       0      33   
...     ...   ...  ...    ...    ...    ...    ...    ...     ...     ...   
20866   398   492  ...    388    306    777   1214    544     596     310   
20867  1777  1664  ...   1342   2152   2684   3016   2507    2922    2181   
20868  1428  1420  ...   1614   2412   1989   3660   1967    3438    2310   
20869   151   120  ...    163    248    212    228    115     176     274   
20870  2938  3611  ...   2750   4389   6393   6123   6007    5885    3232   

       RCFN07  RCFN12                              V62  
0          20       0         ENSG00000227232.5_WASH7P  
1          47      94  ENSG00000241860.7_RP11-34P13.13  
2           0      60         ENSG00000279457.4_WASH9P  
3         181       6    ENSG00000228463.10_AP006222.1  
4           1      43  ENSG00000237094.12_RP4-669L17.4  
...       ...     ...                              ...  
20866     560    1216        ENSG00000212907.2_MT-ND4L  
20867    2247    3114         ENSG00000198886.2_MT-ND4  
20868    2560    2795         ENSG00000198786.2_MT-ND5  
20869     196     264         ENSG00000198695.2_MT-ND6  
20870    4480    7954         ENSG00000198727.2_MT-CYB  

[20871 rows x 62 columns]

In [4]:
TPM_count

Name   Description        FN42       FN45        FN48  \
0       ENSG00000227232.5        WASH7P    0.727646   2.320532    0.122902   
1       ENSG00000279457.4        WASH9P    0.438081   1.018116    0.467328   
2      ENSG00000237094.12  RP4-669L17.4    0.347535   0.255187    0.285115   
3       ENSG00000225630.1      MTND2P28    1.789469   0.648745    0.375206   
4       ENSG00000237973.1      MTCO1P12    3.547817   2.085855    5.109321   
...                   ...           ...         ...        ...         ...   
20760   ENSG00000212907.2       MT-ND4L   52.758490  31.879300   46.375410   
20761   ENSG00000198886.2        MT-ND4   56.855790  31.320310   52.333380   
20762   ENSG00000198786.2        MT-ND5   23.571220  23.024840   35.921590   
20763   ENSG00000198695.2        MT-ND6    9.571104   4.386254    6.715108   
20764   ENSG00000198727.2        MT-CYB  114.331600  71.824720  156.205200   

             FN50       FN51        FN52       FN54        FN56  ...  \
0        0.798591   0.431733    1.531822   1.046038    0.584521  ...   
1        1.758025   2.257250    1.138455   0.429999    0.353596  ...   
2        0.138611   0.304394    0.310385   0.302206    0.453209  ...   
3        0.801972   0.329507    0.921121   1.941946    0.000000  ...   
4        2.588628  35.293741    4.154285   2.775012    2.761343  ...   
...           ...        ...         ...        ...         ...  ...   
20760   53.507600  26.294630   50.505760  51.526300   59.860840  ...   
20761   39.267520  29.649830   48.047430  48.662430   42.824420  ...   
20762   32.868120  21.452840   29.638220  29.587880   27.650680  ...   
20763    6.325959   3.767673   10.426680  10.798450    8.064875  ...   
20764  139.616600  67.986950  127.491700  96.674070  111.665100  ...   

           CFN03       CFN04      CFN05      CFN09       CFN11      RCFN01  \
0       0.424949    1.173461   1.129727   0.000000    0.000000    0.000000   
1       1.586985    0.867613   0.674327   1.622020    1.201449    4.812525   
2       0.127714    0.108732   0.002988   0.126418    0.302067    0.000000   
3       1.158317    0.663413   0.200518   1.677175    1.580437    1.039601   
4       1.460486    3.094965   3.501660   3.296443    4.597010    4.216409   
...          ...         ...        ...        ...         ...         ...   
20760  26.965610   30.450650  46.740810  71.862030   53.364270   61.960200   
20761  19.728390   45.297890  34.152170  37.763540   52.019550   64.255050   
20762  17.952370   38.414240  19.149160  34.673820   30.881320   57.202130   
20763   6.257558   13.632190   7.044494   7.455114    6.231438   10.106900   
20764  48.576210  111.007700  97.744500  92.120730  149.768900  155.498200   

          RCFN06     RCFN07      RCFN12                              V62  
0       0.000000   0.302067    0.000000         ENSG00000227232.5_WASH7P  
1       0.000000   0.000000    0.972005         ENSG00000279457.4_WASH9P  
2       0.131356   0.003435    0.166665  ENSG00000237094.12_RP4-669L17.4  
3       1.870045   0.000000    1.517439       ENSG00000225630.1_MTND2P28  
4       3.705244   3.110327    3.968716       ENSG00000237973.1_MTCO1P12  
...          ...        ...         ...                              ...  
20760  24.844880  38.731240   94.896310        ENSG00000212907.2_MT-ND4L  
20761  36.973570  32.872820   51.403760         ENSG00000198886.2_MT-ND4  
20762  29.629680  28.336950   34.909000         ENSG00000198786.2_MT-ND5  
20763  12.130100   7.488040   11.380420         ENSG00000198695.2_MT-ND6  
20764  65.835320  78.752400  157.766100         ENSG00000198727.2_MT-CYB  

[20765 rows x 62 columns]

## python read count & TPM connt

In [5]:
read_count_s = set(read_count['V62'])
read_count_s

{'ENSG00000214654.8_B3GALT9',
 'ENSG00000234338.1_RP11-797H7.1',
 'ENSG00000241666.2_RP3-455J7.4',
 'ENSG00000166359.11_WDR88',
 'ENSG00000138443.17_ABI2',
 'ENSG00000213967.11_ZNF726',
 'ENSG00000287356.1_RP11-181G12.8',
 'ENSG00000163520.14_FBLN2',
 'ENSG00000204006.10_C1orf185',
 'ENSG00000152583.13_SPARCL1',
 'ENSG00000116761.12_CTH',
 'ENSG00000279821.1_RP11-1334A24.5',
 'ENSG00000086062.13_B4GALT1',
 'ENSG00000166411.14_IDH3A',
 'ENSG00000113240.13_CLK4',
 'ENSG00000120832.10_MTERF2',
 'ENSG00000287190.1_CH17-353B19.2',
 'ENSG00000132664.12_POLR3F',
 'ENSG00000151304.6_SRFBP1',
 'ENSG00000160321.15_ZNF208',
 'ENSG00000204410.16_MSH5',
 'ENSG00000121957.15_GPSM2',
 'ENSG00000231758.3_ARHGAP15-AS1',
 'ENSG00000186951.16_PPARA',
 'ENSG00000167487.12_KLHL26',
 'ENSG00000126070.20_AGO3',
 'ENSG00000132436.12_FIGNL1',
 'ENSG00000170315.14_UBB',
 'ENSG00000123552.18_USP45',
 'ENSG00000100243.21_CYB5R3',
 'ENSG00000198822.11_GRM3',
 'ENSG00000049246.15_PER3',
 'ENSG00000189430.13_NCR1',


In [6]:
TPM_coun_s = set(TPM_count['V62'])
TPM_coun_s

{'ENSG00000214654.8_B3GALT9',
 'ENSG00000234338.1_RP11-797H7.1',
 'ENSG00000241666.2_RP3-455J7.4',
 'ENSG00000166359.11_WDR88',
 'ENSG00000138443.17_ABI2',
 'ENSG00000213967.11_ZNF726',
 'ENSG00000287356.1_RP11-181G12.8',
 'ENSG00000163520.14_FBLN2',
 'ENSG00000204006.10_C1orf185',
 'ENSG00000152583.13_SPARCL1',
 'ENSG00000116761.12_CTH',
 'ENSG00000086062.13_B4GALT1',
 'ENSG00000270269.1_IMMP1LP1',
 'ENSG00000166411.14_IDH3A',
 'ENSG00000113240.13_CLK4',
 'ENSG00000120832.10_MTERF2',
 'ENSG00000287190.1_CH17-353B19.2',
 'ENSG00000132664.12_POLR3F',
 'ENSG00000151304.6_SRFBP1',
 'ENSG00000160321.15_ZNF208',
 'ENSG00000204410.16_MSH5',
 'ENSG00000121957.15_GPSM2',
 'ENSG00000231758.3_ARHGAP15-AS1',
 'ENSG00000186951.16_PPARA',
 'ENSG00000167487.12_KLHL26',
 'ENSG00000126070.20_AGO3',
 'ENSG00000132436.12_FIGNL1',
 'ENSG00000170315.14_UBB',
 'ENSG00000123552.18_USP45',
 'ENSG00000100243.21_CYB5R3',
 'ENSG00000049246.15_PER3',
 'ENSG00000189430.13_NCR1',
 'ENSG00000227066.2_RP3-340N1.2',


In [7]:
we_gene_list = read_count_s & TPM_coun_s
we_gene_list

{'ENSG00000214654.8_B3GALT9',
 'ENSG00000234338.1_RP11-797H7.1',
 'ENSG00000241666.2_RP3-455J7.4',
 'ENSG00000166359.11_WDR88',
 'ENSG00000138443.17_ABI2',
 'ENSG00000213967.11_ZNF726',
 'ENSG00000287356.1_RP11-181G12.8',
 'ENSG00000163520.14_FBLN2',
 'ENSG00000204006.10_C1orf185',
 'ENSG00000152583.13_SPARCL1',
 'ENSG00000116761.12_CTH',
 'ENSG00000086062.13_B4GALT1',
 'ENSG00000166411.14_IDH3A',
 'ENSG00000113240.13_CLK4',
 'ENSG00000120832.10_MTERF2',
 'ENSG00000287190.1_CH17-353B19.2',
 'ENSG00000132664.12_POLR3F',
 'ENSG00000151304.6_SRFBP1',
 'ENSG00000160321.15_ZNF208',
 'ENSG00000204410.16_MSH5',
 'ENSG00000121957.15_GPSM2',
 'ENSG00000231758.3_ARHGAP15-AS1',
 'ENSG00000186951.16_PPARA',
 'ENSG00000167487.12_KLHL26',
 'ENSG00000126070.20_AGO3',
 'ENSG00000132436.12_FIGNL1',
 'ENSG00000170315.14_UBB',
 'ENSG00000123552.18_USP45',
 'ENSG00000100243.21_CYB5R3',
 'ENSG00000049246.15_PER3',
 'ENSG00000189430.13_NCR1',
 'ENSG00000227066.2_RP3-340N1.2',
 'ENSG00000204348.10_DXO',
 'EN

In [8]:
len(we_gene_list)

20109

## R

In [9]:
_r_gene_final = pd.read_csv('final_passed_gene_list.csv')
r_gene_final = _r_gene_final.drop(columns='Unnamed: 0')
r_gene_final

final_passed_gene_list
0             ENSG00000227232.5_WASH7P
1             ENSG00000279457.4_WASH9P
2      ENSG00000237094.12_RP4-669L17.4
3           ENSG00000225630.1_MTND2P28
4           ENSG00000237973.1_MTCO1P12
...                                ...
20104        ENSG00000212907.2_MT-ND4L
20105         ENSG00000198886.2_MT-ND4
20106         ENSG00000198786.2_MT-ND5
20107         ENSG00000198695.2_MT-ND6
20108         ENSG00000198727.2_MT-CYB

[20109 rows x 1 columns]

In [10]:
name = r_gene_final['final_passed_gene_list'].str.split('_')
r_gene_final['gene'] = name.str.get(1)
r_gene_final

final_passed_gene_list          gene
0             ENSG00000227232.5_WASH7P        WASH7P
1             ENSG00000279457.4_WASH9P        WASH9P
2      ENSG00000237094.12_RP4-669L17.4  RP4-669L17.4
3           ENSG00000225630.1_MTND2P28      MTND2P28
4           ENSG00000237973.1_MTCO1P12      MTCO1P12
...                                ...           ...
20104        ENSG00000212907.2_MT-ND4L       MT-ND4L
20105         ENSG00000198886.2_MT-ND4        MT-ND4
20106         ENSG00000198786.2_MT-ND5        MT-ND5
20107         ENSG00000198695.2_MT-ND6        MT-ND6
20108         ENSG00000198727.2_MT-CYB        MT-CYB

[20109 rows x 2 columns]

In [11]:
r_gene_final.duplicated(['gene']).value_counts()

False    20104
True         5
dtype: int64

gene_id는 다른데 gene 이름이 같은 gene이 존제

gene : 20104 <br>
gene_id : 20109


# GTF

In [12]:
gtf = pd.read_csv('gencode.v38.genes.txt', sep='\t', engine='python', encoding='cp949')
gtf

chromosome        type                      gene_id  \
0            chr1        gene  gene_id "ENSG00000223972.5"   
1            chr1  transcript  gene_id "ENSG00000223972.5"   
2            chr1        exon  gene_id "ENSG00000223972.5"   
3            chr1        exon  gene_id "ENSG00000223972.5"   
4            chr1        exon  gene_id "ENSG00000223972.5"   
...           ...         ...                          ...   
465022       chrM  transcript  gene_id "ENSG00000210195.2"   
465023       chrM        exon  gene_id "ENSG00000210195.2"   
465024       chrM        gene  gene_id "ENSG00000210196.2"   
465025       chrM  transcript  gene_id "ENSG00000210196.2"   
465026       chrM        exon  gene_id "ENSG00000210196.2"   

                                              gene_type             gene_name  
0        gene_type "transcribed_unprocessed_pseudogene"   gene_name "DDX11L1"  
1        gene_type "transcribed_unprocessed_pseudogene"   gene_name "DDX11L1"  
2        gene_type "transcribed_unprocessed_pseudogene"   gene_name "DDX11L1"  
3        gene_type "transcribed_unprocessed_pseudogene"   gene_name "DDX11L1"  
4        gene_type "transcribed_unprocessed_pseudogene"   gene_name "DDX11L1"  
...                                                 ...                   ...  
465022                              gene_type "Mt_tRNA"     gene_name "MT-TT"  
465023                              gene_type "Mt_tRNA"     gene_name "MT-TT"  
465024                              gene_type "Mt_tRNA"     gene_name "MT-TP"  
465025                              gene_type "Mt_tRNA"     gene_name "MT-TP"  
465026                              gene_type "Mt_tRNA"     gene_name "MT-TP"  

[465027 rows x 5 columns]

## 파일 정리

In [13]:
v3 = gtf['gene_id'].str.split('"')
v3_1 = gtf['gene_type'].str.split('"')
v3_2 = gtf['gene_name'].str.split('"')
gtf['gene_id'] = v3.str.get(1)
gtf['gene_type'] = v3_1.str.get(1)
gtf['gene_name'] = v3_2.str.get(1)

In [14]:
gtf

chromosome        type            gene_id  \
0            chr1        gene  ENSG00000223972.5   
1            chr1  transcript  ENSG00000223972.5   
2            chr1        exon  ENSG00000223972.5   
3            chr1        exon  ENSG00000223972.5   
4            chr1        exon  ENSG00000223972.5   
...           ...         ...                ...   
465022       chrM  transcript  ENSG00000210195.2   
465023       chrM        exon  ENSG00000210195.2   
465024       chrM        gene  ENSG00000210196.2   
465025       chrM  transcript  ENSG00000210196.2   
465026       chrM        exon  ENSG00000210196.2   

                                 gene_type gene_name  
0       transcribed_unprocessed_pseudogene   DDX11L1  
1       transcribed_unprocessed_pseudogene   DDX11L1  
2       transcribed_unprocessed_pseudogene   DDX11L1  
3       transcribed_unprocessed_pseudogene   DDX11L1  
4       transcribed_unprocessed_pseudogene   DDX11L1  
...                                    ...       ...  
465022                             Mt_tRNA     MT-TT  
465023                             Mt_tRNA     MT-TT  
465024                             Mt_tRNA     MT-TP  
465025                             Mt_tRNA     MT-TP  
465026                             Mt_tRNA     MT-TP  

[465027 rows x 5 columns]

In [15]:
gtf['chromosome'].value_counts()

chr1     43790
chr2     34893
chr3     26791
chr11    25080
chr12    24740
chr17    24459
chr19    23125
chr6     22969
chr7     22930
chr5     22080
chr4     20058
chr16    19810
chr10    18519
chr8     18425
chr9     17968
chr15    17504
chrX     16297
chr14    15783
chr20    10982
chr22    10039
chr13     9915
chr18     9007
chr21     6184
chrY      3568
chrM       111
Name: chromosome, dtype: int64

In [16]:
gtf['type'].value_counts()

exon          348619
gene           58204
transcript     58204
Name: type, dtype: int64

chrM : 미토콘드리아 염색체

## chromosome X, Y 제거 

In [17]:
gtfxy = gtf[(gtf['chromosome'] != 'chrX') & (gtf['chromosome'] != 'chrY')]
gtfxy

chromosome        type            gene_id  \
0            chr1        gene  ENSG00000223972.5   
1            chr1  transcript  ENSG00000223972.5   
2            chr1        exon  ENSG00000223972.5   
3            chr1        exon  ENSG00000223972.5   
4            chr1        exon  ENSG00000223972.5   
...           ...         ...                ...   
465022       chrM  transcript  ENSG00000210195.2   
465023       chrM        exon  ENSG00000210195.2   
465024       chrM        gene  ENSG00000210196.2   
465025       chrM  transcript  ENSG00000210196.2   
465026       chrM        exon  ENSG00000210196.2   

                                 gene_type gene_name  
0       transcribed_unprocessed_pseudogene   DDX11L1  
1       transcribed_unprocessed_pseudogene   DDX11L1  
2       transcribed_unprocessed_pseudogene   DDX11L1  
3       transcribed_unprocessed_pseudogene   DDX11L1  
4       transcribed_unprocessed_pseudogene   DDX11L1  
...                                    ...       ...  
465022                             Mt_tRNA     MT-TT  
465023                             Mt_tRNA     MT-TT  
465024                             Mt_tRNA     MT-TP  
465025                             Mt_tRNA     MT-TP  
465026                             Mt_tRNA     MT-TP  

[445162 rows x 5 columns]

## Exon, gene, transcript 중 Exon, transcript 제거 -> gene만 남기기

In [18]:
gtf_gene = gtfxy[gtfxy['type'] == 'gene']
gtf_gene

chromosome  type            gene_id  \
0            chr1  gene  ENSG00000223972.5   
6            chr1  gene  ENSG00000227232.5   
19           chr1  gene  ENSG00000278267.1   
22           chr1  gene  ENSG00000243485.5   
28           chr1  gene  ENSG00000237613.2   
...           ...   ...                ...   
465012       chrM  gene  ENSG00000198695.2   
465015       chrM  gene  ENSG00000210194.1   
465018       chrM  gene  ENSG00000198727.2   
465021       chrM  gene  ENSG00000210195.2   
465024       chrM  gene  ENSG00000210196.2   

                                 gene_type    gene_name  
0       transcribed_unprocessed_pseudogene      DDX11L1  
6                   unprocessed_pseudogene       WASH7P  
19                                   miRNA    MIR6859-1  
22                                  lncRNA  MIR1302-2HG  
28                                  lncRNA      FAM138A  
...                                    ...          ...  
465012                      protein_coding       MT-ND6  
465015                             Mt_tRNA        MT-TE  
465018                      protein_coding       MT-CYB  
465021                             Mt_tRNA        MT-TT  
465024                             Mt_tRNA        MT-TP  

[55290 rows x 5 columns]

## Protein coding, lncRNA 남기기

In [19]:
len(gtfxy['gene_type'].value_counts())

40

protein_coding                        274012 <br>
lncRNA                                 92341 <br>
processed_pseudogene                   29138 <br>
unprocessed_pseudogene                 11077<br>
transcribed_unprocessed_pseudogene      8849<br>
misc_RNA                                5705<br>
snRNA                                   5248<br>
miRNA                                   4351<br>
TEC                                     3205<br>
snoRNA                                  2251<br>
transcribed_processed_pseudogene        2188<br>
transcribed_unitary_pseudogene          1822<br>
rRNA_pseudogene                         1375<br>
IG_V_pseudogene                          661<br>
IG_V_gene                                578<br>
unitary_pseudogene                       471<br>
TR_V_gene                                423<br>
polymorphic_pseudogene                   334<br>
TR_J_gene                                237<br>
rRNA                                     135<br>
TR_V_pseudogene                          123<br>
scaRNA                                   111<br>
IG_D_gene                                111<br>
IG_C_gene                                 91<br>
Mt_tRNA                                   66<br>
IG_J_gene                                 54<br>
pseudogene                                48<br>
TR_C_gene                                 34<br>
IG_C_pseudogene                           33<br>
ribozyme                                  15<br>
TR_D_gene                                 12<br>
TR_J_pseudogene                           12<br>
sRNA                                      12<br>
translated_unprocessed_pseudogene          9<br>
IG_J_pseudogene                            9<br>
translated_processed_pseudogene            6<br>
Mt_rRNA                                    6<br>
vault_RNA                                  3<br>
IG_pseudogene                              3<br>
scRNA                                      3<br>

In [20]:
gtf_type = gtf_gene[(gtf_gene['gene_type'] == 'protein_coding') | (gtf_gene['gene_type'] == 'lncRNA')]
gtf_type

chromosome  type            gene_id       gene_type      gene_name
22           chr1  gene  ENSG00000243485.5          lncRNA    MIR1302-2HG
28           chr1  gene  ENSG00000237613.2          lncRNA        FAM138A
41           chr1  gene  ENSG00000186092.7  protein_coding          OR4F5
46           chr1  gene  ENSG00000238009.6          lncRNA   RP11-34P13.7
66           chr1  gene  ENSG00000239906.1          lncRNA  RP11-34P13.14
...           ...   ...                ...             ...            ...
464994       chrM  gene  ENSG00000212907.2  protein_coding        MT-ND4L
464997       chrM  gene  ENSG00000198886.2  protein_coding         MT-ND4
465009       chrM  gene  ENSG00000198786.2  protein_coding         MT-ND5
465012       chrM  gene  ENSG00000198695.2  protein_coding         MT-ND6
465018       chrM  gene  ENSG00000198727.2  protein_coding         MT-CYB

[34183 rows x 5 columns]

### 우리 데이터 gene 이름과 동일하게 만들기

In [21]:
gtf_type['gene'] = _gtf_type['gene_id'] + '_' + _gtf_type['gene_name']
gtf_type

NameError: name '_gtf_type' is not defined

### 중복확인

In [ ]:
display(gtf_type[_gtf_type['gene'].duplicated()])
print(len(gtf_type[_gtf_type['gene'].duplicated()]))

Empty DataFrame
Columns: [chromosome, type, gene_id, gene_type, gene_name, gene]
Index: []

0


In [ ]:
len(gtf_type[gtf_type['chromosome'] == 'chrM'])

13

## gtf_gene_list 뽑이오기

In [ ]:
gtf_gene_list = set(gtf_type['gene'])
gtf_gene_list

{'ENSG00000196578.4_OR5AC2',
 'ENSG00000255555.1_AP000857.3',
 'ENSG00000229278.2_RP11-483F11.7',
 'ENSG00000266513.1_RP11-874J12.3',
 'ENSG00000234883.6_MIR155HG',
 'ENSG00000158955.11_WNT9B',
 'ENSG00000167642.13_SPINT2',
 'ENSG00000197312.12_DDI2',
 'ENSG00000170953.4_OR8B12',
 'ENSG00000205363.7_INSYN1',
 'ENSG00000244128.7_LINC01322',
 'ENSG00000280485.1_RP11-437J2.4',
 'ENSG00000276707.1_RP11-445F12.2',
 'ENSG00000185002.10_RFX6',
 'ENSG00000261126.7_RBFADN',
 'ENSG00000261642.1_RP11-309H21.3',
 'ENSG00000269802.1_CTB-180A7.3',
 'ENSG00000166226.13_CCT2',
 'ENSG00000229334.1_AC046143.3',
 'ENSG00000185379.21_RAD51D',
 'ENSG00000231061.1_LINC00395',
 'ENSG00000254731.1_CTD-2005H7.1',
 'ENSG00000272954.1_KB-1440D3.13',
 'ENSG00000163235.16_TGFA',
 'ENSG00000273448.1_RP11-166O4.6',
 'ENSG00000236292.1_USH2A-AS1',
 'ENSG00000251488.1_RP11-404I7.2',
 'ENSG00000268287.1_CTB-60B18.18',
 'ENSG00000166986.15_MARS1',
 'ENSG00000253293.5_HOXA10',
 'ENSG00000288070.1_RP11-217B19.1',
 'ENSG00

# final gene & GTF 

In [ ]:
## 우리 데이터 gene 이름 
print(we_gene_list)
print(gtf_gene_list)

{'ENSG00000156500.16_PABIR3', 'ENSG00000234883.6_MIR155HG', 'ENSG00000167642.13_SPINT2', 'ENSG00000197312.12_DDI2', 'ENSG00000205363.7_INSYN1', 'ENSG00000122121.12_XPNPEP2', 'ENSG00000261126.7_RBFADN', 'ENSG00000166226.13_CCT2', 'ENSG00000229334.1_AC046143.3', 'ENSG00000185379.21_RAD51D', 'ENSG00000163235.16_TGFA', 'ENSG00000273448.1_RP11-166O4.6', 'ENSG00000166986.15_MARS1', 'ENSG00000253293.5_HOXA10', 'ENSG00000160211.19_G6PD', 'ENSG00000138613.14_APH1B', 'ENSG00000161940.11_BCL6B', 'ENSG00000174276.7_ZNHIT2', 'ENSG00000112624.12_BICRAL', 'ENSG00000135829.17_DHX9', 'ENSG00000112852.7_PCDHB2', 'ENSG00000266777.1_SH3GL1P1', 'ENSG00000087152.16_ATXN7L3', 'ENSG00000231535.7_LINC00278', 'ENSG00000242615.1_CTC-359D24.3', 'ENSG00000242372.9_EIF6', 'ENSG00000174004.6_NRROS', 'ENSG00000196387.10_ZNF140', 'ENSG00000170899.11_GSTA4', 'ENSG00000265298.1_RP13-104F24.3', 'ENSG00000141385.11_AFG3L2', 'ENSG00000144857.15_BOC', 'ENSG00000251455.1_RP11-164P12.3', 'ENSG00000231132.1_RP11-40C11.2', 'ENS

In [ ]:
final_gene_list = we_gene_list & gtf_gene_list
print(final_gene_list)
print(len(final_gene_list))

{'ENSG00000138660.12_AP1AR', 'ENSG00000137818.12_RPLP1', 'ENSG00000286194.1_RP11-848P1.10', 'ENSG00000124786.12_SLC35B3', 'ENSG00000261526.2_CTB-31O20.2', 'ENSG00000234883.6_MIR155HG', 'ENSG00000188176.12_SMTNL2', 'ENSG00000141384.13_TAF4B', 'ENSG00000184117.12_NIPSNAP1', 'ENSG00000167642.13_SPINT2', 'ENSG00000120068.7_HOXB8', 'ENSG00000035141.8_FAM136A', 'ENSG00000287355.1_RP11-27J8.4', 'ENSG00000170619.10_COMMD5', 'ENSG00000197312.12_DDI2', 'ENSG00000241370.6_RPP21', 'ENSG00000177738.4_CTD-2201E18.3', 'ENSG00000197279.5_ZNF165', 'ENSG00000205363.7_INSYN1', 'ENSG00000123191.16_ATP7B', 'ENSG00000182359.15_KBTBD3', 'ENSG00000239857.7_GET4', 'ENSG00000226332.2_RP11-157P1.4', 'ENSG00000137198.10_GMPR', 'ENSG00000237975.7_FLG-AS1', 'ENSG00000166292.12_TMEM100', 'ENSG00000140563.15_MCTP2', 'ENSG00000233559.2_LINC00513', 'ENSG00000226026.6_RP11-57H12.3', 'ENSG00000129028.9_THAP10', 'ENSG00000261126.7_RBFADN', 'ENSG00000006025.12_OSBPL7', 'ENSG00000112294.14_ALDH5A1', 'ENSG00000188171.16_ZNF6

In [ ]:
r_gene = set(r_gene_final['final_passed_gene_list'])

In [ ]:
check_gene = final_gene_list & r_gene
len(check_gene)

17268

# 우리 데이터에서 미토콘드리아 염섹체 찾기

In [ ]:
we_gtf = gtf_type[gtf_type['gene'].isin(list(final_gene_list))]
we_gtf

chromosome  type             gene_id       gene_type  gene_name  \
259          chr1  gene  ENSG00000237491.10          lncRNA  LINC01409   
279          chr1  gene   ENSG00000177757.2          lncRNA     FAM87B   
284          chr1  gene  ENSG00000228794.10          lncRNA  LINC01128   
340          chr1  gene  ENSG00000187634.13  protein_coding     SAMD11   
358          chr1  gene  ENSG00000188976.11  protein_coding      NOC2L   
...           ...   ...                 ...             ...        ...   
464994       chrM  gene   ENSG00000212907.2  protein_coding    MT-ND4L   
464997       chrM  gene   ENSG00000198886.2  protein_coding     MT-ND4   
465009       chrM  gene   ENSG00000198786.2  protein_coding     MT-ND5   
465012       chrM  gene   ENSG00000198695.2  protein_coding     MT-ND6   
465018       chrM  gene   ENSG00000198727.2  protein_coding     MT-CYB   

                                gene  
259     ENSG00000237491.10_LINC01409  
279         ENSG00000177757.2_FAM87B  
284     ENSG00000228794.10_LINC01128  
340        ENSG00000187634.13_SAMD11  
358         ENSG00000188976.11_NOC2L  
...                              ...  
464994     ENSG00000212907.2_MT-ND4L  
464997      ENSG00000198886.2_MT-ND4  
465009      ENSG00000198786.2_MT-ND5  
465012      ENSG00000198695.2_MT-ND6  
465018      ENSG00000198727.2_MT-CYB  

[17268 rows x 6 columns]

In [ ]:
display(we_gtf[we_gtf['chromosome'] == 'chrM'])
print(len(we_gtf[we_gtf['chromosome'] == 'chrM']))

chromosome  type            gene_id       gene_type gene_name  \
464931       chrM  gene  ENSG00000198888.2  protein_coding    MT-ND1   
464943       chrM  gene  ENSG00000198763.3  protein_coding    MT-ND2   
464961       chrM  gene  ENSG00000198804.2  protein_coding    MT-CO1   
464970       chrM  gene  ENSG00000198712.1  protein_coding    MT-CO2   
464976       chrM  gene  ENSG00000228253.1  protein_coding   MT-ATP8   
464979       chrM  gene  ENSG00000198899.2  protein_coding   MT-ATP6   
464982       chrM  gene  ENSG00000198938.2  protein_coding    MT-CO3   
464988       chrM  gene  ENSG00000198840.2  protein_coding    MT-ND3   
464994       chrM  gene  ENSG00000212907.2  protein_coding   MT-ND4L   
464997       chrM  gene  ENSG00000198886.2  protein_coding    MT-ND4   
465009       chrM  gene  ENSG00000198786.2  protein_coding    MT-ND5   
465012       chrM  gene  ENSG00000198695.2  protein_coding    MT-ND6   
465018       chrM  gene  ENSG00000198727.2  protein_coding    MT-CYB   

                             gene  
464931   ENSG00000198888.2_MT-ND1  
464943   ENSG00000198763.3_MT-ND2  
464961   ENSG00000198804.2_MT-CO1  
464970   ENSG00000198712.1_MT-CO2  
464976  ENSG00000228253.1_MT-ATP8  
464979  ENSG00000198899.2_MT-ATP6  
464982   ENSG00000198938.2_MT-CO3  
464988   ENSG00000198840.2_MT-ND3  
464994  ENSG00000212907.2_MT-ND4L  
464997   ENSG00000198886.2_MT-ND4  
465009   ENSG00000198786.2_MT-ND5  
465012   ENSG00000198695.2_MT-ND6  
465018   ENSG00000198727.2_MT-CYB

13


## DEG에서 미토콘드리아 gene이 있는지 확인

In [ ]:
CD_Con = pd.read_csv('CD_Con.txt', sep=',', engine='python', encoding='cp949')
CrF_Con = pd.read_csv('CrF_Con.txt', sep=',', engine='python', encoding='cp949')
CrF_CD = pd.read_csv('CrF_CD.txt', sep=',', engine='python', encoding='cp949')

CD_Con = CD_Con.rename(columns={'Unnamed: 0' : 'gene'})
CrF_Con = CrF_Con.rename(columns={'Unnamed: 0' : 'gene'})
CrF_CD = CrF_CD.rename(columns={'Unnamed: 0' : 'gene'})

CD_Con_deg = CD_Con[(CD_Con['padj'] < 0.05) & (abs(CD_Con['log2FoldChange']) > 1.5)]
CrF_Con_deg = CrF_Con[(CrF_Con['padj'] < 0.05) & (abs(CrF_Con['log2FoldChange']) > 1.5)]
CrF_CD_deg = CrF_CD[(CrF_CD['padj'] < 0.05) & (abs(CrF_CD['log2FoldChange']) > 1.5)]

display(CD_Con_deg)
display(CrF_Con_deg)
display(CrF_CD_deg)

In [ ]:
display(gtf_type[gtf_type['gene'].isin(list(CD_Con_deg['gene']))])
display(gtf_type[gtf_type['gene'].isin(list(CrF_Con_deg['gene']))])
display(gtf_type[gtf_type['gene'].isin(list(CrF_CD_deg['gene']))])

chromosome  type             gene_id       gene_type  gene_name  \
5509         chr1  gene  ENSG00000188257.12  protein_coding    PLA2G2A   
5689         chr1  gene  ENSG00000117245.13  protein_coding      KIF17   
14989        chr1  gene  ENSG00000134709.12  protein_coding      HOOK1   
21664        chr1  gene  ENSG00000121933.19  protein_coding     TMIGD3   
22263        chr1  gene  ENSG00000116748.22  protein_coding      AMPD1   
...           ...   ...                 ...             ...        ...   
430233      chr21  gene  ENSG00000154639.19  protein_coding      CXADR   
431269      chr21  gene  ENSG00000171189.17  protein_coding      GRIK1   
433518      chr21  gene  ENSG00000160185.15  protein_coding    UBASH3A   
434377      chr21  gene   ENSG00000227039.6          lncRNA  ITGB2-AS1   
438072      chr22  gene  ENSG00000099958.15  protein_coding      DERL3   

                               gene  
5509     ENSG00000188257.12_PLA2G2A  
5689       ENSG00000117245.13_KIF17  
14989      ENSG00000134709.12_HOOK1  
21664     ENSG00000121933.19_TMIGD3  
22263      ENSG00000116748.22_AMPD1  
...                             ...  
430233     ENSG00000154639.19_CXADR  
431269     ENSG00000171189.17_GRIK1  
433518   ENSG00000160185.15_UBASH3A  
434377  ENSG00000227039.6_ITGB2-AS1  
438072     ENSG00000099958.15_DERL3  

[130 rows x 6 columns]

chromosome  type             gene_id       gene_type   gene_name  \
568          chr1  gene  ENSG00000186891.14  protein_coding    TNFRSF18   
575          chr1  gene  ENSG00000186827.11  protein_coding     TNFRSF4   
724          chr1  gene   ENSG00000169962.5  protein_coding      TAS1R3   
2765         chr1  gene   ENSG00000231789.2          lncRNA  PIK3CD-AS2   
2855         chr1  gene  ENSG00000162444.12  protein_coding        RBP7   
...           ...   ...                 ...             ...         ...   
438321      chr22  gene  ENSG00000128271.22  protein_coding     ADORA2A   
438911      chr22  gene   ENSG00000225783.8          lncRNA        MIAT   
439963      chr22  gene  ENSG00000183963.19  protein_coding        SMTN   
441245      chr22  gene  ENSG00000100385.14  protein_coding       IL2RB   
442818      chr22  gene  ENSG00000167077.13  protein_coding        MEI1   

                                gene  
568      ENSG00000186891.14_TNFRSF18  
575       ENSG00000186827.11_TNFRSF4  
724         ENSG00000169962.5_TAS1R3  
2765    ENSG00000231789.2_PIK3CD-AS2  
2855         ENSG00000162444.12_RBP7  
...                              ...  
438321    ENSG00000128271.22_ADORA2A  
438911        ENSG00000225783.8_MIAT  
439963       ENSG00000183963.19_SMTN  
441245      ENSG00000100385.14_IL2RB  
442818       ENSG00000167077.13_MEI1  

[463 rows x 6 columns]

chromosome  type             gene_id       gene_type     gene_name  \
4488         chr1  gene  ENSG00000142632.17  protein_coding      ARHGEF19   
6605         chr1  gene   ENSG00000188822.8  protein_coding          CNR2   
14989        chr1  gene  ENSG00000134709.12  protein_coding         HOOK1   
17759        chr1  gene  ENSG00000153898.13  protein_coding        MCOLN2   
21417        chr1  gene   ENSG00000259834.1          lncRNA  RP11-284N8.3   
...           ...   ...                 ...             ...           ...   
423293      chr20  gene  ENSG00000061656.11  protein_coding         SPAG4   
425383      chr20  gene   ENSG00000100985.7  protein_coding          MMP9   
428134      chr20  gene  ENSG00000101194.18  protein_coding       SLC17A9   
438072      chr22  gene  ENSG00000099958.15  protein_coding         DERL3   
442818      chr22  gene  ENSG00000167077.13  protein_coding          MEI1   

                                  gene  
4488       ENSG00000142632.17_ARHGEF19  
6605            ENSG00000188822.8_CNR2  
14989         ENSG00000134709.12_HOOK1  
17759        ENSG00000153898.13_MCOLN2  
21417   ENSG00000259834.1_RP11-284N8.3  
...                                ...  
423293        ENSG00000061656.11_SPAG4  
425383          ENSG00000100985.7_MMP9  
428134      ENSG00000101194.18_SLC17A9  
438072        ENSG00000099958.15_DERL3  
442818         ENSG00000167077.13_MEI1  

[101 rows x 6 columns]

---

In [ ]:
_non_list = gene_list - final_gene_list

In [ ]:
gtf

V1          V2            gene_id  \
0       chr1        gene  ENSG00000223972.5   
1       chr1  transcript  ENSG00000223972.5   
2       chr1        exon  ENSG00000223972.5   
3       chr1        exon  ENSG00000223972.5   
4       chr1        exon  ENSG00000223972.5   
...      ...         ...                ...   
465022  chrM  transcript  ENSG00000210195.2   
465023  chrM        exon  ENSG00000210195.2   
465024  chrM        gene  ENSG00000210196.2   
465025  chrM  transcript  ENSG00000210196.2   
465026  chrM        exon  ENSG00000210196.2   

                                 gene_type gene_name  
0       transcribed_unprocessed_pseudogene   DDX11L1  
1       transcribed_unprocessed_pseudogene   DDX11L1  
2       transcribed_unprocessed_pseudogene   DDX11L1  
3       transcribed_unprocessed_pseudogene   DDX11L1  
4       transcribed_unprocessed_pseudogene   DDX11L1  
...                                    ...       ...  
465022                             Mt_tRNA     MT-TT  
465023                             Mt_tRNA     MT-TT  
465024                             Mt_tRNA     MT-TP  
465025                             Mt_tRNA     MT-TP  
465026                             Mt_tRNA     MT-TP  

[465027 rows x 5 columns]

In [ ]:
non_list = list(_non_list)

In [ ]:
non_list

['YES1P1',
 'SNX2P1',
 'RP1-131F15.2',
 'CTB-78O21.1',
 'RP11-403I13.4',
 'USP12P2',
 'YBX1P3',
 'AFF2',
 'RPL21P119',
 'APEX2',
 'SLC25A39P1',
 'RPS20P31',
 'NKRF',
 'RP3-425C14.4',
 'RPLP0P6',
 'RPS19P3',
 'RAB9A',
 'BMS1P10',
 'RP11-115L11.1',
 'BNIP3P26',
 'PFN1P2',
 'PNPLA4',
 'PCDHGB9P',
 'RP11-532F6.2',
 'PDLIM1P4',
 'EEF1A1P10',
 'GAPDHP61',
 'RP11-449J21.3',
 'RN7SKP80',
 'CTD-2014D20.1',
 'RP13-554M15.7',
 'NCAPGP1',
 'RPS15AP13',
 'TRGC1',
 'CTD-2083E4.5',
 'TRBV2',
 'MSL3',
 'RPL32P18',
 'CNTNAP3P2',
 'RBMS1P1',
 'DOCK11',
 'IGLV1-44',
 'RPL7P21',
 'RP11-536C12.1',
 'PRPS1',
 'SMIM10',
 'RPL7P49',
 'RPL14P1',
 'RP11-135N5.3',
 'ROCK1P1',
 'IGKV2D-28',
 'RPL17P50',
 'SNORA74A',
 'TCEAL2',
 'RPL32P32',
 'SNORA80A',
 'KRT223P',
 'NR0B1',
 'IGHV2-70',
 'RPL5P1',
 'BCYRN1',
 'IGKV3D-11',
 'TXLNG',
 'JADE3',
 'SNRPGP4',
 'RP11-404O13.4',
 'HERC2P9',
 'PABPC1P1',
 'RP1-261D10.1',
 'ZNF711',
 'RP11-686D22.11',
 'HYDIN2',
 'UBQLN2',
 'BMS1P2',
 'CFL1P3',
 'YWHABP2',
 'RAB11FIP1P1',


### 제거된 gene을 gtf 파일에서 찾기 

In [ ]:
non_gtf = gtf[gtf['gene_name'].isin(non_list)]

In [ ]:
non_gtf_gene = non_gtf[non_gtf['V2'] == 'gene']
non_gtf_gene

V1    V2             gene_id                           gene_type  \
6       chr1  gene   ENSG00000227232.5              unprocessed_pseudogene   
96      chr1  gene   ENSG00000279457.4              unprocessed_pseudogene   
135     chr1  gene  ENSG00000237094.12  transcribed_unprocessed_pseudogene   
203     chr1  gene   ENSG00000225630.1              unprocessed_pseudogene   
206     chr1  gene   ENSG00000237973.1              unprocessed_pseudogene   
...      ...   ...                 ...                                 ...   
463511  chrY  gene  ENSG00000198692.10                      protein_coding   
464115  chrY  gene   ENSG00000251925.1                              snoRNA   
464771  chrY  gene   ENSG00000251796.1                              snoRNA   
464919  chrM  gene   ENSG00000211459.2                             Mt_rRNA   
464925  chrM  gene   ENSG00000210082.2                             Mt_rRNA   

           gene_name  
6             WASH7P  
96            WASH9P  
135     RP4-669L17.4  
203         MTND2P28  
206         MTCO1P12  
...              ...  
463511        EIF1AY  
464115       SNORA70  
464771       SNORA70  
464919       MT-RNR1  
464925       MT-RNR2  

[3098 rows x 5 columns]

In [ ]:
non_gtf_gene[non_gtf_gene['V1'] == 'chrX']

V1    V2             gene_id             gene_type      gene_name
445105  chrX  gene   ENSG00000275287.5              misc_RNA    Metazoa_SRP
445315  chrX  gene  ENSG00000124343.14        protein_coding             XG
445328  chrX  gene  ENSG00000056998.20        protein_coding           GYG2
445347  chrX  gene  ENSG00000006756.16        protein_coding           ARSD
445417  chrX  gene   ENSG00000101825.8        protein_coding          MXRA5
...      ...   ...                 ...                   ...            ...
461214  chrX  gene  ENSG00000155962.13        protein_coding          CLIC2
461224  chrX  gene   ENSG00000225008.1  processed_pseudogene         TWF1P2
461227  chrX  gene   ENSG00000228986.1  processed_pseudogene        PHF10P1
461234  chrX  gene   ENSG00000224216.1                lncRNA  RP13-228J13.1
461267  chrX  gene  ENSG00000185973.12        protein_coding          TMLHE

[646 rows x 5 columns]

In [ ]:
read_count[read_count['Description'] == 'ARSD']

Name Description  FN42  FN45  FN48  FN50  FN51  FN52  \
20170  ENSG00000006756.16        ARSD  1205  1844   744  1363   682   978   

       FN54  FN56  ...  CFN03  CFN04  CFN05  CFN09  CFN11  RCFN01  RCFN06  \
20170   946  2182  ...   1919   1341    989   1538    729    1630     786   

       RCFN07  RCFN12                      V62  
20170    1468    1754  ENSG00000006756.16_ARSD  

[1 rows x 62 columns]

In [ ]:
list(gene_list) == 'TMLHE'

False

In [ ]:
gtf_gene['gene_type'].value_counts()

protein_coding                        18393
lncRNA                                15790
processed_pseudogene                   9189
unprocessed_pseudogene                 2270
misc_RNA                               1898
snRNA                                  1747
miRNA                                  1441
TEC                                     995
transcribed_unprocessed_pseudogene      860
snoRNA                                  748
rRNA_pseudogene                         458
transcribed_processed_pseudogene        449
IG_V_pseudogene                         187
IG_V_gene                               145
transcribed_unitary_pseudogene          136
TR_V_gene                               106
unitary_pseudogene                       91
TR_J_gene                                79
polymorphic_pseudogene                   48
rRNA                                     45
scaRNA                                   37
IG_D_gene                                37
TR_V_pseudogene                 

In [ ]:
gtf_gene[gtf_gene['gene_type'] == 'processed_pseudogene']['gene_name']

56               CICP27
60        RP11-34P13.15
63        RP11-34P13.16
126           RPL23AP24
158             WBP1LP7
              ...      
444221     RP11-191L9.5
444316          RPL35P8
444399      RP3-522J7.5
444407    RP11-494O16.3
445027       AC002056.3
Name: gene_name, Length: 9189, dtype: object

In [ ]:
read_count[read_count['Description'].isin(gtf_gene[gtf_gene['gene_type'] == 'snRNA']['gene_name'])]

Name Description  FN42  FN45  FN48  FN50  FN51  FN52  \
204    ENSG00000206652.1      RNU1-1    69    44    24    48    52    39   
213    ENSG00000207005.1      RNU1-2    71   127    58    54   179    90   
589    ENSG00000199377.1     RNU5F-1     9     1    11     0    47     6   
590    ENSG00000200169.1     RNU5D-1     6    14    10    12    29     3   
1119   ENSG00000273727.1          U1    13     3     4    12    15     7   
1126   ENSG00000277610.1     RNVU1-4    42    18    34    15    27    25   
1136   ENSG00000206737.1    RNVU1-18    94    55    43    11    96    84   
1143   ENSG00000207205.1    RNVU1-15    91   138    67    92   107    66   
1150   ENSG00000270722.1    RNVU1-31     2    12    16    12    76    20   
1174   ENSG00000273768.1    RNVU1-29    80   156   113   171   326   271   
1176   ENSG00000274428.1    RNVU1-25    18    12     4    12    45     5   
1179   ENSG00000286172.1     RNVU1-8     0     0    20    20    21    12   
1191   ENSG00000206585.1     RNVU1-7    39    23     9    60    57    21   
1194   ENSG00000199879.1    RNVU1-22     0     6    14     7    47    23   
1196   ENSG00000238825.1     RNVU1-2     6     4    11    13    75    25   
1198   ENSG00000201183.1     RNVU1-3     0     3     1    22    20     3   
1201   ENSG00000274210.1    RNVU1-27   156   121   247   273   386   247   
1210   ENSG00000206828.1    RNVU1-30    78    74    51   103   162    94   
2671   ENSG00000222724.1    RNU2-63P    17     7    37    44   118    63   
2907   ENSG00000264229.1    RNU4ATAC     3    61    33    50   110    53   
10252  ENSG00000221676.1    RNU6ATAC     0     4    14     8     0    15   
13272  ENSG00000202538.1      RNU4-2  1087   706  2342  1721   916  1889   
13273  ENSG00000200795.1      RNU4-1   229   112   240   345   222   333   
14936  ENSG00000199568.1     RNU5A-1    57    16    19    33   101    62   
14937  ENSG00000200156.1     RNU5B-1    12    22    20    22    18    28   

       FN54  FN56  ...  CFN03  CFN04  CFN05  CFN09  CFN11  RCFN01  RCFN06  \
204      57    12  ...     45     24     73    110     80       0      15   
213      60    85  ...    113     30     88    239     67      52      70   
589      12    19  ...      0      0      0     10     14       0       0   
590      19     0  ...      0     37      0     12      0       0       0   
1119      5     4  ...      0      0     14     21      0      88       0   
1126     55    17  ...     34     22     73    178     43       0       0   
1136    126    72  ...    103     39    190    207     20      35     122   
1143    116    80  ...    280     27    155    411     63      52     100   
1150      8    21  ...      9      0     21    205     13       0      52   
1174    100   176  ...     99     70    263    460     67     153      62   
1176     13    31  ...      8      0     19     39      0       0       0   
1179     16    80  ...     35     10     10    112      6       0      60   
1191     22    29  ...     27     32     47     98     25       0      24   
1194     11     0  ...      0     39     30     26      0       5       0   
1196      7    78  ...     20      1     24    146      0      28      38   
1198      3    21  ...      0     15     24     54     23       0       0   
1201    363   244  ...    552    186    520   1139    253     328     410   
1210     85   119  ...     83     38    101    337     36      93     185   
2671     71    87  ...     18     81     42    190     29      32     104   
2907     82   101  ...     53     53     76    275     80      75      76   
10252    16    12  ...      8      0     38     22     11      34      12   
13272  1373  1498  ...   2937   2004   4814   8892   7471    2843    3130   
13273   315   245  ...    443    353    935   1335    645     392     502   
14936    81    65  ...     63     38     87    304    105       0      76   
14937    28    37  ...     29     28     21     92    123      62      40   

       RCFN07  RCFN12                   

In [ ]:
df = pd.read_csv('Sample information_59_CrFCDCon.txt', sep='\t', engine='python', encoding='cp949')
df

Sample    Number Sex Type  bulk
0     FN42  60040459   M   CD     1
1     FN45  43194243   F   CD     1
2     FN48  60941103   M   CD     1
3     FN50  44507417   F   CD     1
4     FN51  34247282   M   CD     1
5     FN52  61115743   F   CD     1
6     FN54  37518248   M   CD     1
7     FN56  61176786   F   CD     1
8     FN57  61209352   F   CD     1
9     FN58  61236518   M   CD     1
10    FN60  15436524   M   CD     1
11    FN61  61341674   F   CD     1
12    FN62  61285596   M   CD     1
13    FN63  61233324   M   CD     1
14    FN65  61198180   M   CD     1
15    FN67  61358007   M   CD     1
16    FN68  54297173   M   CD     1
17    FN69  11648527   M   CD     0
18    FN71  55116145   M   CD     1
19    FN72  61712430   M   CD     0
20    FN74  48511999   M   CD     0
21    FN77  40703763   F   CD     1
22    FN82  43200016   M   CD     1
23    FP42  60040459   M  CrF     1
24    FP45  43194243   F  CrF     1
25    FP48  60941103   M  CrF     1
26    FP50  44507417   F  CrF     1
27    FP51  34247282   M  CrF     1
28    FP52  61115743   F  CrF     1
29    FP54  37518248   M  CrF     1
30    FP56  61176786   F  CrF     1
31    FP57  61209352   F  CrF     1
32    FP58  61236518   M  CrF     1
33    FP60  15436524   M  CrF     1
34    FP61  61341674   F  CrF     1
35    FP62  61285596   M  CrF     1
36    FP63  61233324   M  CrF     1
37    FP65  61198180   M  CrF     1
38    FP67  61358007   M  CrF     1
39    FP68  54297173   M  CrF     1
40    FP69  11648527   M  CrF     0
41    FP71  55116145   M  CrF     1
42    FP72  61712430   M  CrF     0
43    FP74  48511999   M  CrF     0
44    FP77  40703763   F  CrF     1
45    FP82  43200016   M  CrF     1
46   UC100  40237576   F  Con     0
47    UC92  17526771   F  Con     0
48    UC94  40647205   F  Con     0
49   CFN02  61228269   F  Con     1
50   CFN03  61284478   F  Con     1
51   CFN04  61420786   F  Con     1
52   CFN05  20193595   M  Con     1
53   CFN09  60194419   F  Con     1
54   CFN11  60382425   M  Con     0
55  RCFN01  61187027   M  Con     1
56  RCFN06  61500790   M  Con     1
57  RCFN07  13726562   F  Con     1
58  RCFN12  62158488   F  Con     0

In [ ]:
print('CD : ', len(df[df['Type'] == 'CD']))
print('Con : ', len(df[df['Type'] == 'Con']))
print('CrF : ', len(df[df['Type'] == 'CrF']))

CD :  23
Con :  13
CrF :  23
